### PR TITLE
Minor changes for save data utility

### PR DESCRIPTION
**Pull request by Dangles91**
Finished cellSaveDataGetListItem implementation
Implemented cellSaveDataUserGetListItem
-- this will help games like RDR and Minecraft load correctly

Minor fixes to save_op
Added comments for future UI behaviours
Removed PFuncDone callback from save_op, not used or required.

### CODE REVIEW

**Comment by kamer1337** [comment]
nice not sure if its this commit but it fixed tony hawk's proving ground load save game

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 1]
```diff
@@ -83,6 +83,11 @@ static NEVER_INLINE s32 savedata_op(ppu_thread& ppu, u32 operation, u32 version,
 	vm::ptr<CellSaveDataFileSet>  fileSet  = g_savedata_context.ptr(&savedata_context::fileSet);
 
 	// path of the specified user (00000001 by default)
+	if (userId == CELL_SYSUTIL_USERID_CURRENT)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: It's already checked for zero later

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 2]
```diff
@@ -296,6 +314,10 @@ static NEVER_INLINE s32 savedata_op(ppu_thread& ppu, u32 operation, u32 version,
 			}
 			case CELL_SAVEDATA_FOCUSPOS_NEWDATA:
 			{
+				//focus new data, placed at the end of the vector.
+				// not currently supported in the ui
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Wrong comments are missing the point, new data is focused on -1 value.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 3]
```diff
@@ -60,10 +61,9 @@ std::mutex g_savedata_mutex;
 
 static NEVER_INLINE s32 savedata_op(ppu_thread& ppu, u32 operation, u32 version, vm::cptr<char> dirName,
 	u32 errDialog, PSetList setList, PSetBuf setBuf, PFuncList funcList, PFuncFixed funcFixed, PFuncStat funcStat,
-	PFuncFile funcFile, u32 container, u32 unknown, vm::ptr<void> userdata, u32 userId, PFuncDone funcDone)
+	PFuncFile funcFile, u32 container, u32 unknown, vm::ptr<void> userdata, u32 userId)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Nope, this megafunction was modelled after an existing one. The arguments, their positions were preserved.

**Comment by Luffykun007** [comment]
Fixed Doki-Doki Universe,Before you would get stuck on this screen.

capturar3

Now go ingame.

capturar

capturar1

capturar2

**Comment by Zangetsu38** [comment]
oh good :D

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp (hunk 2 above)
Comment (Dangles91) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: This was emulate adding a new data item to the list. Instead we use a new save data button.

See - https://youtu.be/_f0kC7lgdtM?t=46s

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp (hunk 1 above)
Comment (Dangles91) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Would you prefer to remove the step for retrieving the current ID, where the TODO is. Or remove the check for 0 in the string initialisation?

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp (hunk 3 above)
Comment (Dangles91) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: I'll chuck this back in on the next commit

File: rpcs3/Emu/Cell/Modules/cellSysutil.h [hunk 4]
```diff
@@ -79,6 +79,8 @@ enum
 	CELL_SYSUTIL_SYSCHAT_STOP                    = 0x0162,
 	CELL_SYSUTIL_SYSCHAT_VOICE_STREAMING_RESUMED = 0x0163,
 	CELL_SYSUTIL_SYSCHAT_VOICE_STREAMING_PAUSED  = 0x0164,
+
+	CELL_SYSUTIL_USERID_CURRENT = 0,
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSysutil.h]: This repeats definition in another file. Just use zeros everywhere, it doesn't matter.

**Comment by Zangetsu38** [comment]
also fixed this game :D
image

**Comment by Nekotekina** [comment]
Stop spamming with all these things.

**Pull request merged**